app/query_objects/cafe_queries.rb

```diff
@@ -0,0 +1,54 @@
+class CafeQueries
+  # returns: all users who worked during study period(lp) and year
+  def self.working_users(lp, year)
+    join_cafe_shifts(User.joins(:cafe_shifts), lp, year).includes(:cafe_workers).uniq
+  end
+
+  # returns: all cafe_workers from study period and year
+  def self.cafe_workers(lp, year)
+    join_cafe_shifts(CafeWorker.joins(:cafe_shift), lp, year)
+  end
+
+  def self.highscore(lp, year)
+    join_cafe_shifts(User.select('users.id, users.firstname, users.lastname, count(*) as score')
+                     .joins(:cafe_shifts), lp, year)
+    .group('user_id')
+    .order('score desc')
+    .limit(10)
+  end
+
+  def self.free_shifts(lp, year)
+    CafeShift.where(lp: lp)
+    .where('start > ? and year(start) = ?', Time.zone.now, year.year)
+    .without_worker
+  end
+
+  def self.for_day(day)
+    between(day.beginning_of_day, day.end_of_day)
+  end
+
+  def self.between(start, stop)
+    CafeShift.where('start BETWEEN ? AND ?', start, stop)
+    .order(pass: :asc)
+    .includes(:user)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Use 2 (not 0) spaces for indenting an expression spanning multiple lines.